bugbear.py

```diff
@@ -68,11 +68,15 @@ def gen_line_based_checks(self):
 
         The following simple checks are based on the raw lines, not the AST.
         """
+        noqa_type_ignore_regex = re.compile(r"\s*#\s*(noqa:|type:\s*ignore).*$")
         for lineno, line in enumerate(self.lines, start=1):
             # Special case: ignore long shebang (following pycodestyle).
             if lineno == 1 and line.startswith("#!"):
                 continue
 
+            # At first, removing noqa and type: ignore trailing comments"
+            line = noqa_type_ignore_regex.sub("", line)
+
             length = len(line) - 1
```

Comment:
```suggestion
            no_comment_line = noqa_type_ignore_regex.sub("", line)

            length = len(no_comment_line) - 1
```
Let's maybe be more explicit in variable name to make it clearer to people in the future reading this ...